### ISSUE TITLE
Explanation of NUnit1013 appears to be inverted

### ISSUE THREAD

**Issue by SeanKilleen** [nunit/nunit.analyzers#207]
When documenting the analyzers as part of #206, I noticed that the expectation for the rule appears to be the inverse of what it is actually checking.

The explanation in the docs and the analyzer itself reads:

> Async test method must have non-generic Task return type when no result is expected.

However, what it actually means is that it must have a *generic*, _not_ a non-generic, Task type.

The below code will cause `NUnit1013` to be thrown, showing that it's the explanation that's incorrect, not the check itself:

```csharp
[TestCase(1)]
public async Task<string> Nunit1013SampleTest(int numberValue)
{

    return await ConvertNumber(numberValue);
}

public Task<string> ConvertNumber(int numberValue)
{
    return Task.FromResult(numberValue.ToString());
}
```

I'll add the appropriate explanation in #206 but then will add a follow-up PR to fix this in the codebase & docs.

**Comment by SeanKilleen** [comment]
Wow, I'm sleepy. This is totally correct. I read non-generic to mean `Task<T>` instead of `Task`. It was, in fact, my own brain that was inverted! 😆

**Comment by mikkelbu** [comment]
No problem 😄. I also had to check a couple of times to be certain that I had not inverted the statement.
